### ISSUE TITLE
Document RPC Calls

### ISSUE THREAD

**Issue by wpank** [substrate-developer-hub/knowledgebase#56]
People looking through the documentation may notice that RPC documentation is missing from the knowledge base. I'm aware [they are referenced here from the autogeneration](https://polkadot.js.org/api/substrate/rpc.html), but at least referencing them somewhere in the knowledge base would be useful. (A step further would be giving examples of the inputs/outputs of these, as the above link doesn't make this too explicit and can be a point of confusion).

**Comment by joepetrowski** [comment]
We discussed this recently in chat. There should probably be an autogenerated set of RPC reference docs that we publish somewhere. For the KB, I think RPC documentation should perhaps have a basics page, but the primary content should really be across multiple areas:

- RPC Basics: basic `curl` examples, protocols supported, list of tools
- Node management
- Defining runtime RPC endpoints (e.g. `TransactionPayment.queryInfo`)
- Using RPC for storage (already in storage documentation)
- Decoding RPC data (already in SCALE)

**Comment by JoshOrndorff** [comment]
> Defining runtime RPC endpoints (e.g. TransactionPayment.queryInfo)

For crosslinking purposes: https://substrate.dev/recipes/3-entrees/custom-rpc.html

Also, I recently had this conversation

Fred
> runtime-specific RPCs was (and is) an anti-pattern

Me
> Ah, I didn't realize that.

Fred
> The thing is, maybe the framework isn't good enough to let people do things through the generic APIs, and people are resorting to runtime-specific RPCs because they have no better option then it's on us to provider a better option

**Comment by joepetrowski** [comment]
Yeah, I agree with Fred here, especially given the recent weights issue where the host couldn't talk to the runtime. So we might not document "how to create it" but perhaps "how to troubleshoot it".

**Comment by danforbes** [comment]
Ref: https://github.com/w3f/PSPs/pull/6, https://github.com/w3f/PSPs/blob/psp-rpc-api/psp-002.md
